In [1]:
%pip install pandas numpy

Note: you may need to restart the kernel to use updated packages.



[notice] A new release of pip is available: 26.1.2 -> 26.2.1
[notice] To update, run: python.exe -m pip install --upgrade pip


In [2]:
import pandas as pd
import numpy as np

print("Pandas and NumPy are working!")

Pandas and NumPy are working!


In [3]:
inflation_df = pd.read_csv("inflation.csv")

print("Inflation dataset loaded successfully!")

Inflation dataset loaded successfully!


In [4]:
poverty_df = pd.read_csv("poverty.csv")

print("Poverty dataset loaded successfully!")

Poverty dataset loaded successfully!


In [5]:
print("Inflation data:")
display(inflation_df.head())

print("Poverty data:")
display(poverty_df.head())

Inflation data:


,FREQ,FREQ_LABEL,REF_AREA,REF_AREA_LABEL,INDICATOR,INDICATOR_LABEL,UNIT_MEASURE,UNIT_MEASURE_LABEL,DATABASE_ID,DATABASE_ID_LABEL,...,2020,2021,2022,2023,2024,2025,2026,2027,2028,2029
0,A,Annual,ABW,Aruba,IMF_WEO_PCPIPCH,"Inflation, average consumer prices, Percent ch...",PC,Percentage change,IMF_WEO,World Economic Outlook (WEO),...,-1.313,0.744,5.52,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1,A,Annual,ABW,Aruba,IMF_WEO_PCPIPCH,"Inflation, average consumer prices, Percent ch...",PC,Percentage change,IMF_WEO,World Economic Outlook (WEO),...,NaN,NaN,NaN,3.366,2.345,1.815,1.914,1.998,1.998,1.998
2,A,Annual,AFG,Afghanistan,IMF_WEO_PCPIPCH,"Inflation, average consumer prices, Percent ch...",PC,Percentage change,IMF_WEO,World Economic Outlook (WEO),...,5.607,7.762,10.60,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3,A,Annual,AGO,Angola,IMF_WEO_PCPIPCH,"Inflation, average consumer prices, Percent ch...",PC,Percentage change,IMF_WEO,World Economic Outlook (WEO),...,22.277,25.765,21.36,13.639,NaN,NaN,NaN,NaN,NaN,NaN
4,A,Annual,AGO,Angola,IMF_WEO_PCPIPCH,"Inflation, average consumer prices, Percent ch...",PC,Percentage change,IMF_WEO,World Economic Outlook (WEO),...,NaN,NaN,NaN,NaN,22.004,12.818,8.506,7.442,7.442,7.442


Poverty data:


,STRUCTURE,STRUCTURE_ID,ACTION,FREQ,REF_AREA,INDICATOR,SEX,AGE,URBANISATION,UNIT_MEASURE,...,2016,2017,2018,2019,2020,2021,2022,2023,2024,2025
0,datastructure,WB.DATA360:DS_DATA360(1.3),I,A,XKX,WB_WDI_SI_POV_UMIC,_T,_T,_T,PT_POP,...,53.4,49.5,54.7,38.3,38.9,47.0,38.2,NaN,NaN,NaN
1,datastructure,WB.DATA360:DS_DATA360(1.3),I,A,MOZ,WB_WDI_SI_POV_UMIC,_T,_T,_T,PT_POP,...,NaN,NaN,NaN,96.5,NaN,NaN,96.3,NaN,NaN,NaN
2,datastructure,WB.DATA360:DS_DATA360(1.3),I,A,EAS,WB_WDI_SI_POV_UMIC,_T,_T,_T,PT_POP,...,40.7,38.7,35.8,33.5,34.1,29.2,29.6,27.6,26.4,NaN
3,datastructure,WB.DATA360:DS_DATA360(1.3),I,A,GMB,WB_WDI_SI_POV_UMIC,_T,_T,_T,PT_POP,...,NaN,NaN,NaN,NaN,78.2,NaN,NaN,NaN,NaN,NaN
4,datastructure,WB.DATA360:DS_DATA360(1.3),I,A,TUN,WB_WDI_SI_POV_UMIC,_T,_T,_T,PT_POP,...,NaN,NaN,NaN,NaN,NaN,20.7,NaN,NaN,NaN,NaN


In [6]:
print("Inflation columns:")
print(inflation_df.columns.tolist())

print("\nPoverty columns:")
print(poverty_df.columns.tolist())

Inflation columns:
['FREQ', 'FREQ_LABEL', 'REF_AREA', 'REF_AREA_LABEL', 'INDICATOR', 'INDICATOR_LABEL', 'UNIT_MEASURE', 'UNIT_MEASURE_LABEL', 'DATABASE_ID', 'DATABASE_ID_LABEL', 'UNIT_MULT', 'UNIT_MULT_LABEL', 'OBS_STATUS', 'OBS_STATUS_LABEL', 'OBS_CONF', 'OBS_CONF_LABEL', '1980', '1981', '1982', '1983', '1984', '1985', '1986', '1987', '1988', '1989', '1990', '1991', '1992', '1993', '1994', '1995', '1996', '1997', '1998', '1999', '2000', '2001', '2002', '2003', '2004', '2005', '2006', '2007', '2008', '2009', '2010', '2011', '2012', '2013', '2014', '2015', '2016', '2017', '2018', '2019', '2020', '2021', '2022', '2023', '2024', '2025', '2026', '2027', '2028', '2029']

Poverty columns:
['STRUCTURE', 'STRUCTURE_ID', 'ACTION', 'FREQ', 'REF_AREA', 'INDICATOR', 'SEX', 'AGE', 'URBANISATION', 'UNIT_MEASURE', 'COMP_BREAKDOWN_1', 'COMP_BREAKDOWN_2', 'COMP_BREAKDOWN_3', 'AGG_METHOD', 'UNIT_TYPE', 'DECIMALS', 'DATABASE_ID', 'TIME_FORMAT', 'UNIT_MULT', 'OBS_STATUS', 'OBS_CONF', 'FREQ_LABEL', 'REF_AR

In [7]:
print("Inflation dataset:")
display(inflation_df.head())

print("\nPoverty dataset:")
display(poverty_df.head())

Inflation dataset:


,FREQ,FREQ_LABEL,REF_AREA,REF_AREA_LABEL,INDICATOR,INDICATOR_LABEL,UNIT_MEASURE,UNIT_MEASURE_LABEL,DATABASE_ID,DATABASE_ID_LABEL,...,2020,2021,2022,2023,2024,2025,2026,2027,2028,2029
0,A,Annual,ABW,Aruba,IMF_WEO_PCPIPCH,"Inflation, average consumer prices, Percent ch...",PC,Percentage change,IMF_WEO,World Economic Outlook (WEO),...,-1.313,0.744,5.52,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1,A,Annual,ABW,Aruba,IMF_WEO_PCPIPCH,"Inflation, average consumer prices, Percent ch...",PC,Percentage change,IMF_WEO,World Economic Outlook (WEO),...,NaN,NaN,NaN,3.366,2.345,1.815,1.914,1.998,1.998,1.998
2,A,Annual,AFG,Afghanistan,IMF_WEO_PCPIPCH,"Inflation, average consumer prices, Percent ch...",PC,Percentage change,IMF_WEO,World Economic Outlook (WEO),...,5.607,7.762,10.60,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3,A,Annual,AGO,Angola,IMF_WEO_PCPIPCH,"Inflation, average consumer prices, Percent ch...",PC,Percentage change,IMF_WEO,World Economic Outlook (WEO),...,22.277,25.765,21.36,13.639,NaN,NaN,NaN,NaN,NaN,NaN
4,A,Annual,AGO,Angola,IMF_WEO_PCPIPCH,"Inflation, average consumer prices, Percent ch...",PC,Percentage change,IMF_WEO,World Economic Outlook (WEO),...,NaN,NaN,NaN,NaN,22.004,12.818,8.506,7.442,7.442,7.442



Poverty dataset:


,STRUCTURE,STRUCTURE_ID,ACTION,FREQ,REF_AREA,INDICATOR,SEX,AGE,URBANISATION,UNIT_MEASURE,...,2016,2017,2018,2019,2020,2021,2022,2023,2024,2025
0,datastructure,WB.DATA360:DS_DATA360(1.3),I,A,XKX,WB_WDI_SI_POV_UMIC,_T,_T,_T,PT_POP,...,53.4,49.5,54.7,38.3,38.9,47.0,38.2,NaN,NaN,NaN
1,datastructure,WB.DATA360:DS_DATA360(1.3),I,A,MOZ,WB_WDI_SI_POV_UMIC,_T,_T,_T,PT_POP,...,NaN,NaN,NaN,96.5,NaN,NaN,96.3,NaN,NaN,NaN
2,datastructure,WB.DATA360:DS_DATA360(1.3),I,A,EAS,WB_WDI_SI_POV_UMIC,_T,_T,_T,PT_POP,...,40.7,38.7,35.8,33.5,34.1,29.2,29.6,27.6,26.4,NaN
3,datastructure,WB.DATA360:DS_DATA360(1.3),I,A,GMB,WB_WDI_SI_POV_UMIC,_T,_T,_T,PT_POP,...,NaN,NaN,NaN,NaN,78.2,NaN,NaN,NaN,NaN,NaN
4,datastructure,WB.DATA360:DS_DATA360(1.3),I,A,TUN,WB_WDI_SI_POV_UMIC,_T,_T,_T,PT_POP,...,NaN,NaN,NaN,NaN,NaN,20.7,NaN,NaN,NaN,NaN


In [8]:
south_africa_inflation = inflation_df[
    inflation_df["REF_AREA"] == "ZAF"
]

display(south_africa_inflation)

,FREQ,FREQ_LABEL,REF_AREA,REF_AREA_LABEL,INDICATOR,INDICATOR_LABEL,UNIT_MEASURE,UNIT_MEASURE_LABEL,DATABASE_ID,DATABASE_ID_LABEL,...,2020,2021,2022,2023,2024,2025,2026,2027,2028,2029
381,A,Annual,ZAF,South Africa,IMF_WEO_PCPIPCH,"Inflation, average consumer prices, Percent ch...",PC,Percentage change,IMF_WEO,World Economic Outlook (WEO),...,3.275,4.556,6.869,5.9,NaN,NaN,NaN,NaN,NaN,NaN
382,A,Annual,ZAF,South Africa,IMF_WEO_PCPIPCH,"Inflation, average consumer prices, Percent ch...",PC,Percentage change,IMF_WEO,World Economic Outlook (WEO),...,NaN,NaN,NaN,NaN,4.863,4.5,4.5,4.5,4.5,4.5


In [9]:
south_africa_poverty = poverty_df[
    poverty_df["REF_AREA"] == "ZAF"
]

display(south_africa_poverty)

,STRUCTURE,STRUCTURE_ID,ACTION,FREQ,REF_AREA,INDICATOR,SEX,AGE,URBANISATION,UNIT_MEASURE,...,2016,2017,2018,2019,2020,2021,2022,2023,2024,2025
62,datastructure,WB.DATA360:DS_DATA360(1.3),I,A,ZAF,WB_WDI_SI_POV_UMIC,_T,_T,_T,PT_POP,...,NaN,NaN,NaN,NaN,NaN,NaN,59.6,NaN,NaN,NaN


In [10]:
print("Inflation columns:")
print(south_africa_inflation.columns.tolist())

Inflation columns:
['FREQ', 'FREQ_LABEL', 'REF_AREA', 'REF_AREA_LABEL', 'INDICATOR', 'INDICATOR_LABEL', 'UNIT_MEASURE', 'UNIT_MEASURE_LABEL', 'DATABASE_ID', 'DATABASE_ID_LABEL', 'UNIT_MULT', 'UNIT_MULT_LABEL', 'OBS_STATUS', 'OBS_STATUS_LABEL', 'OBS_CONF', 'OBS_CONF_LABEL', '1980', '1981', '1982', '1983', '1984', '1985', '1986', '1987', '1988', '1989', '1990', '1991', '1992', '1993', '1994', '1995', '1996', '1997', '1998', '1999', '2000', '2001', '2002', '2003', '2004', '2005', '2006', '2007', '2008', '2009', '2010', '2011', '2012', '2013', '2014', '2015', '2016', '2017', '2018', '2019', '2020', '2021', '2022', '2023', '2024', '2025', '2026', '2027', '2028', '2029']


In [11]:
print("Poverty columns:")
print(south_africa_poverty.columns.tolist())

Poverty columns:
['STRUCTURE', 'STRUCTURE_ID', 'ACTION', 'FREQ', 'REF_AREA', 'INDICATOR', 'SEX', 'AGE', 'URBANISATION', 'UNIT_MEASURE', 'COMP_BREAKDOWN_1', 'COMP_BREAKDOWN_2', 'COMP_BREAKDOWN_3', 'AGG_METHOD', 'UNIT_TYPE', 'DECIMALS', 'DATABASE_ID', 'TIME_FORMAT', 'UNIT_MULT', 'OBS_STATUS', 'OBS_CONF', 'FREQ_LABEL', 'REF_AREA_LABEL', 'INDICATOR_LABEL', 'SEX_LABEL', 'AGE_LABEL', 'URBANISATION_LABEL', 'UNIT_MEASURE_LABEL', 'COMP_BREAKDOWN_1_LABEL', 'COMP_BREAKDOWN_2_LABEL', 'COMP_BREAKDOWN_3_LABEL', 'AGG_METHOD_LABEL', 'UNIT_TYPE_LABEL', 'DECIMALS_LABEL', 'DATABASE_ID_LABEL', 'TIME_FORMAT_LABEL', 'UNIT_MULT_LABEL', 'OBS_STATUS_LABEL', 'OBS_CONF_LABEL', '1963', '1964', '1965', '1966', '1967', '1968', '1969', '1970', '1971', '1972', '1973', '1974', '1975', '1976', '1977', '1978', '1979', '1980', '1981', '1982', '1983', '1984', '1985', '1986', '1987', '1988', '1989', '1990', '1991', '1992', '1993', '1994', '1995', '1996', '1997', '1998', '1999', '2000', '2001', '2002', '2003', '2004', '2005

In [12]:
print(
    south_africa_inflation[
        ["REF_AREA_LABEL", "INDICATOR_LABEL", "OBS_STATUS", "OBS_STATUS_LABEL"]
    ]
)

    REF_AREA_LABEL                                    INDICATOR_LABEL  \
381   South Africa  Inflation, average consumer prices, Percent ch...   
382   South Africa  Inflation, average consumer prices, Percent ch...   

    OBS_STATUS OBS_STATUS_LABEL  
381          A     Normal value  
382          F   Forecast value  


# SELECT THE ACTUAL INFLATION DATA

In [14]:
actual_inflation = south_africa_inflation[
    south_africa_inflation["OBS_STATUS_LABEL"] == "Normal value"
]

display(actual_inflation)

,FREQ,FREQ_LABEL,REF_AREA,REF_AREA_LABEL,INDICATOR,INDICATOR_LABEL,UNIT_MEASURE,UNIT_MEASURE_LABEL,DATABASE_ID,DATABASE_ID_LABEL,...,2020,2021,2022,2023,2024,2025,2026,2027,2028,2029
381,A,Annual,ZAF,South Africa,IMF_WEO_PCPIPCH,"Inflation, average consumer prices, Percent ch...",PC,Percentage change,IMF_WEO,World Economic Outlook (WEO),...,3.275,4.556,6.869,5.9,NaN,NaN,NaN,NaN,NaN,NaN


# EXTRACT THE INFLATION VALUES

In [15]:
years = ["1993", "2000", "2005", "2008", "2010", "2014", "2022"]

In [16]:
inflation_values = actual_inflation[years].iloc[0].astype(float).to_numpy()

print("Inflation values:")
print(inflation_values)

Inflation values:
[ 9.72   5.322  3.367 11.02   4.211  6.104  6.869]


# EXTRACT THE POVERTY VALUES

In [17]:
poverty_values = south_africa_poverty[years].iloc[0].astype(float).to_numpy()

print("Poverty values:")
print(poverty_values)

Poverty values:
[75.4 78.5 74.9 70.5 65.1 65.7 59.6]


# THE NUMPY ARRAYS

In [18]:
years_array = np.array([1993, 2000, 2005, 2008, 2010, 2014, 2022])

inflation_array = np.array(inflation_values)

poverty_array = np.array(poverty_values)

print("Years array:")
print(years_array)

print("\nInflation array:")
print(inflation_array)

print("\nPoverty array:")
print(poverty_array)

Years array:
[1993 2000 2005 2008 2010 2014 2022]

Inflation array:
[ 9.72   5.322  3.367 11.02   4.211  6.104  6.869]

Poverty array:
[75.4 78.5 74.9 70.5 65.1 65.7 59.6]


# THE ARRAY SHAPE

In [19]:
print("Years shape:", years_array.shape)
print("Inflation shape:", inflation_array.shape)
print("Poverty shape:", poverty_array.shape)

Years shape: (7,)
Inflation shape: (7,)
Poverty shape: (7,)


## CALCULATING DESCRIPTIVE STATISTICS

# Inflation Statistics

In [20]:
print("INFLATION STATISTICS")

print("Mean:", np.mean(inflation_array))
print("Median:", np.median(inflation_array))
print("Minimum:", np.min(inflation_array))
print("Maximum:", np.max(inflation_array))
print("Standard deviation:", np.std(inflation_array))

INFLATION STATISTICS
Mean: 6.659
Median: 6.104
Minimum: 3.367
Maximum: 11.02
Standard deviation: 2.6010356179249614


# Poverty Statistics

In [21]:
print("POVERTY STATISTICS")

print("Mean:", np.mean(poverty_array))
print("Median:", np.median(poverty_array))
print("Minimum:", np.min(poverty_array))
print("Maximum:", np.max(poverty_array))
print("Standard deviation:", np.std(poverty_array))

POVERTY STATISTICS
Mean: 69.95714285714286
Median: 70.5
Minimum: 59.6
Maximum: 78.5
Standard deviation: 6.282824010826694


In [22]:
print("INFLATION STATISTICS")

print("Mean:", round(np.mean(inflation_array), 2))
print("Median:", round(np.median(inflation_array), 2))
print("Minimum:", round(np.min(inflation_array), 2))
print("Maximum:", round(np.max(inflation_array), 2))
print("Standard deviation:", round(np.std(inflation_array), 2))

print("\nPOVERTY STATISTICS")

print("Mean:", round(np.mean(poverty_array), 2))
print("Median:", round(np.median(poverty_array), 2))
print("Minimum:", round(np.min(poverty_array), 2))
print("Maximum:", round(np.max(poverty_array), 2))
print("Standard deviation:", round(np.std(poverty_array), 2))

INFLATION STATISTICS
Mean: 6.66
Median: 6.1
Minimum: 3.37
Maximum: 11.02
Standard deviation: 2.6

POVERTY STATISTICS
Mean: 69.96
Median: 70.5
Minimum: 59.6
Maximum: 78.5
Standard deviation: 6.28


# THE YEAR WITH THE HIGHEST INFLATION

In [23]:
highest_inflation_index = np.argmax(inflation_array)

print("Highest inflation year:", years_array[highest_inflation_index])
print("Highest inflation value:", inflation_array[highest_inflation_index])

Highest inflation year: 2008
Highest inflation value: 11.02


# THE YEAR WITH THE LOWEST INFLATION

In [24]:
lowest_inflation_index = np.argmin(inflation_array)

print("Lowest inflation year:", years_array[lowest_inflation_index])
print("Lowest inflation value:", inflation_array[lowest_inflation_index])

Lowest inflation year: 2005
Lowest inflation value: 3.367


# THE YEAR WITH THE HIGHEST POVERTY

In [41]:
highest_poverty_index = np.argmax(poverty_array)

print("Highest poverty year:", years_array[highest_poverty_index])
print("Highest poverty value:", poverty_array[highest_poverty_index])

Highest poverty year: 2000
Highest poverty value: 78.5


# THE YEAR WITH THE LOWEST POVERTY

In [26]:
lowest_poverty_index = np.argmin(poverty_array)

print("Lowest poverty year:", years_array[lowest_poverty_index])
print("Lowest poverty value:", poverty_array[lowest_poverty_index])

Lowest poverty year: 2022
Lowest poverty value: 59.6


# ARRAY OPERATIONS

In [27]:
inflation_change = np.diff(inflation_array)

print("Change in inflation between selected years:")
print(inflation_change)

Change in inflation between selected years:
[-4.398 -1.955  7.653 -6.809  1.893  0.765]


In [28]:
poverty_change = np.diff(poverty_array)

print("Change in poverty between selected years:")
print(poverty_change)

Change in poverty between selected years:
[ 3.1 -3.6 -4.4 -5.4  0.6 -6.1]


# RESHAPING THE NUMPY ARRAYS

In [29]:
years_reshaped = years_array.reshape(1, 7)
inflation_reshaped = inflation_array.reshape(1, 7)
poverty_reshaped = poverty_array.reshape(1, 7)

print("Years reshaped:")
print(years_reshaped)

print("\nInflation reshaped:")
print(inflation_reshaped)

print("\nPoverty reshaped:")
print(poverty_reshaped)

Years reshaped:
[[1993 2000 2005 2008 2010 2014 2022]]

Inflation reshaped:
[[ 9.72   5.322  3.367 11.02   4.211  6.104  6.869]]

Poverty reshaped:
[[75.4 78.5 74.9 70.5 65.1 65.7 59.6]]


In [30]:
print("Years shape:", years_reshaped.shape)
print("Inflation shape:", inflation_reshaped.shape)
print("Poverty shape:", poverty_reshaped.shape)

Years shape: (1, 7)
Inflation shape: (1, 7)
Poverty shape: (1, 7)


In [31]:
correlation_matrix = np.corrcoef(
    inflation_array,
    poverty_array
)

print(correlation_matrix)

[[1.         0.02982964]
 [0.02982964 1.        ]]


In [32]:
correlation = correlation_matrix[0, 1]

print("Correlation between inflation and poverty:", round(correlation, 2))

Correlation between inflation and poverty: 0.03


# A SUMMARY TABLE

In [33]:
summary = pd.DataFrame({
    "Year": years_array,
    "Inflation (%)": inflation_array,
    "Poverty (%)": poverty_array
})

display(summary)

,Year,Inflation (%),Poverty (%)
0,1993,9.720,75.4
1,2000,5.322,78.5
2,2005,3.367,74.9
3,2008,11.020,70.5
4,2010,4.211,65.1
5,2014,6.104,65.7
6,2022,6.869,59.6


# THE DESCRIPTIVE STATISTICS REVIEW

In [34]:
print("FINAL DESCRIPTIVE STATISTICS")
print()

print("INFLATION")
print("Mean:", round(np.mean(inflation_array), 2))
print("Median:", round(np.median(inflation_array), 2))
print("Minimum:", round(np.min(inflation_array), 2))
print("Maximum:", round(np.max(inflation_array), 2))
print("Standard deviation:", round(np.std(inflation_array), 2))

print()

print("POVERTY")
print("Mean:", round(np.mean(poverty_array), 2))
print("Median:", round(np.median(poverty_array), 2))
print("Minimum:", round(np.min(poverty_array), 2))
print("Maximum:", round(np.max(poverty_array), 2))
print("Standard deviation:", round(np.std(poverty_array), 2))

FINAL DESCRIPTIVE STATISTICS

INFLATION
Mean: 6.66
Median: 6.1
Minimum: 3.37
Maximum: 11.02
Standard deviation: 2.6

POVERTY
Mean: 69.96
Median: 70.5
Minimum: 59.6
Maximum: 78.5
Standard deviation: 6.28


# THE HIGHEST AND LOWEST VALUES

In [35]:
print("INFLATION")
print(
    "Highest:",
    years_array[np.argmax(inflation_array)],
    "with",
    np.max(inflation_array),
    "%"
)

print(
    "Lowest:",
    years_array[np.argmin(inflation_array)],
    "with",
    np.min(inflation_array),
    "%"
)

print()

print("POVERTY")
print(
    "Highest:",
    years_array[np.argmax(poverty_array)],
    "with",
    np.max(poverty_array),
    "%"
)

print(
    "Lowest:",
    years_array[np.argmin(poverty_array)],
    "with",
    np.min(poverty_array),
    "%"
)

INFLATION
Highest: 2008 with 11.02 %
Lowest: 2005 with 3.367 %

POVERTY
Highest: 2000 with 78.5 %
Lowest: 2022 with 59.6 %


In [37]:
print("Inflation changes between selected years:")

for i in range(len(inflation_change)):
    print(
        years_array[i],
        "to",
        years_array[i + 1],
        ":",
        round(inflation_change[i], 2),
        "percentage points"
    )

print()

print("Poverty changes between selected years:")

for i in range(len(poverty_change)):
    print(
        years_array[i],
        "to",
        years_array[i + 1],
        ":",
        round(poverty_change[i], 2),
        "percentage points"
    )

Inflation changes between selected years:
1993 to 2000 : -4.4 percentage points
2000 to 2005 : -1.96 percentage points
2005 to 2008 : 7.65 percentage points
2008 to 2010 : -6.81 percentage points
2010 to 2014 : 1.89 percentage points
2014 to 2022 : 0.76 percentage points

Poverty changes between selected years:
1993 to 2000 : 3.1 percentage points
2000 to 2005 : -3.6 percentage points
2005 to 2008 : -4.4 percentage points
2008 to 2010 : -5.4 percentage points
2010 to 2014 : 0.6 percentage points
2014 to 2022 : -6.1 percentage points


# ARRAY RESHAPING

In [38]:
print("Original inflation shape:", inflation_array.shape)

inflation_matrix = inflation_array.reshape(1, 7)

print("Reshaped inflation shape:", inflation_matrix.shape)
print("Reshaped inflation array:")
print(inflation_matrix)

Original inflation shape: (7,)
Reshaped inflation shape: (1, 7)
Reshaped inflation array:
[[ 9.72   5.322  3.367 11.02   4.211  6.104  6.869]]


In [39]:
print("Original poverty shape:", poverty_array.shape)

poverty_matrix = poverty_array.reshape(1, 7)

print("Reshaped poverty shape:", poverty_matrix.shape)
print("Reshaped poverty array:")
print(poverty_matrix)

Original poverty shape: (7,)
Reshaped poverty shape: (1, 7)
Reshaped poverty array:
[[75.4 78.5 74.9 70.5 65.1 65.7 59.6]]


## EXPLAINING MY FINDINGS 

The results show that inflation in South Africa changed over the selected years. The average inflation rate was 6.66%, with the highest rate of 11.02% in 2008 and the lowest rate of 3.37% in 2005.

The poverty rate was higher in the earlier years but generally decreased over time. The highest poverty rate was 78.5% in 2000, while the lowest was 59.6% in 2022.

The correlation between inflation and poverty was 0.03, which shows a very weak relationship between the two variables. This means that inflation and poverty did not consistently increase or decrease together in the selected years.